src/components/Sketches/components/EditSketchModal.js

```diff
@@ -0,0 +1,323 @@
+import React from "react";
+import ReactModal from "react-modal";
+import { Button } from "reactstrap";
+import DropdownButton from "./DropdownButton";
+import { SketchThumbnailArray, LanguageDropdownValues } from "../constants";
+import { Container, Row, Col, FormGroup, Label, Input } from "reactstrap";
+import * as fetch from "../../../lib/fetch.js";
+
+class EditSketchModal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      newLanguage: -1,
+      newName: -1,
+      newThumbnail: -1,
+      spinner: false,
+      error: "",
+      onThumbnails: false,
+    };
+  }
+  componentWillMount() {}
+  componentDidUpdate() {}
+
+  closeModal = () => {
+    if (this.props.onClose && {}.toString.call(this.props.onClose) === "[object Function]") {
+      this.props.onClose();
+    }
+    this.setState({
+      next: false,
+      newLanguage: -1,
+      newName: -1,
+      newThumbnail: -1,
+      error: "",
+      onThumbnails: false,
+      spinner: false,
+    });
+  };
+
+  // Next two bad____input are copied (and slightly modified) from create sketch modal
+
+  badNameInput = () => {
+    if (this.state.newName === -1) {
+      return false;
+    }
+    if (!this.state.newName) {
+      this.setState({ error: "Name is required" });
+      return true;
+    }
+    if (this.state.newName.length > 15) {
+      this.setState({ error: "Name must be 15 characters or less" });
+      return true;
+    }
+    return false;
+  };
+
+  badLanguageInput = () => {
+    if (this.state.newLanguage === -1) {
+      return false;
+    }
+    if (!this.state.newLanguage) {
+      this.setState({ error: "Please select a language" });
+      return true;
+    }
+
+    let notFound = true;
+    for (var i = 0; i < LanguageDropdownValues.length; i++) {
+      if (
+        this.state.newLanguage.display === LanguageDropdownValues[i].display &&
+        this.state.newLanguage.value === LanguageDropdownValues[i].value
+      ) {
+        notFound = false;
+        break;
+      }
+    }
+    if (notFound) {
+      this.setState({ error: "Invalid language selected" });
+      return true;
+    }
+
+    return false;
+  };
+
+  handleSubmitEdit = async e => {
+    e.preventDefault();
+
+    if (this.badNameInput() || this.badLanguageInput()) {
+      // note it's impossible to have a bad thumbnail input
+      return;
+    }
+
+    let data = {};
+    let doUpdate = false;
+
+    if (this.state.newLanguage !== -1) {
+      data["language"] = this.state.newLanguage.value;
```

Comment:
is there a reason for the ["string"] notation? Or just not a fan of . notation?

Reply:
Nah, no reason. Swapped it over.